src/internal/io.js

```diff
@@ -2,26 +2,26 @@ import { IO, SELF_CANCELLATION } from './symbols'
 import { delay as delayUtil, is, identity, check, createSetContextWarning } from './utils'
 import { takeEveryHelper, takeLatestHelper, throttleHelper } from './sagaHelpers'
 
-const TAKE = 'TAKE'
-const PUT = 'PUT'
-const ALL = 'ALL'
-const RACE = 'RACE'
-const CALL = 'CALL'
-const CPS = 'CPS'
-const FORK = 'FORK'
-const JOIN = 'JOIN'
-const CANCEL = 'CANCEL'
-const SELECT = 'SELECT'
-const ACTION_CHANNEL = 'ACTION_CHANNEL'
-const CANCELLED = 'CANCELLED'
-const FLUSH = 'FLUSH'
-const GET_CONTEXT = 'GET_CONTEXT'
-const SET_CONTEXT = 'SET_CONTEXT'
+export const TAKE = 'TAKE'
+export const PUT = 'PUT'
+export const ALL = 'ALL'
+export const RACE = 'RACE'
+export const CALL = 'CALL'
+export const CPS = 'CPS'
+export const FORK = 'FORK'
+export const JOIN = 'JOIN'
+export const CANCEL = 'CANCEL'
+export const SELECT = 'SELECT'
+export const ACTION_CHANNEL = 'ACTION_CHANNEL'
+export const CANCELLED = 'CANCELLED'
+export const FLUSH = 'FLUSH'
+export const GET_CONTEXT = 'GET_CONTEXT'
+export const SET_CONTEXT = 'SET_CONTEXT'
 
 const TEST_HINT =
   '\n(HINT: if you are getting this errors in tests, consider using createMockTask from redux-saga/utils)'
 
-const effect = (type, payload) => ({ [IO]: true, [type]: payload })
+const effect = (type, payload) => ({ [IO]: type, [type]: payload })
```

Comment:
this is a refactoring I wanted to do long time ago.

BTW why dynamic `[type]` property and not just `type` and `payload` properties ? (would also makes inspection easier in devtools) . the purpose of `[IO]` was to avoid name clashes as the user could yield any object so needed

Reply:
hi @yelouafi ! 👋 

if we remove `IO` won't it create the same problem? How today's situation is different in terms of name clashes?

Reply:
also - without `IO` its hard to determine if the object is an effect for the purpose of introduced `is.effect`, hard to distinguish it from any other `{ type, payload }` objects, its a popular shape i.e. for redux actions

Reply:
OTOH we could do `{ [IO]: type, payload }` instead of the dynamic `type` property

Reply:
I meant 3 props : 

```js
{
  [IO]: true  //as a signature
  type: TAKE,
  payload: ...
}
```

then it could be consumed like

```js
if(obj[IO] === true) {
  switch(obj.type) {
    case TAKE:
    ...
  }
}
```

the first test could maybe be put behind a little function `isEffect(obj)`

Reply:
Yeah, that would totally work. Gonna play with it later. Thanks for the input, it's nice to know that you still have an eye on the project 👍 

I've started to refactor a little bit to extract things (like effect runners) from the `proc`'s closure - but as things, especially those runners, access closure's variables I've started to look into what they access and I will try to gather those things under single saga's `state` variable - this potentially could unlock possibility to create custom effects in the user land.